cedar-policy-core/src/ast/value.rs

```diff
@@ -484,38 +485,49 @@ impl std::fmt::Display for ValueKind {
                 fast,
                 authoritative,
             }) => {
-                match authoritative.len() {
-                    0 => write!(f, "[]"),
-                    n @ 1..=5 => {
-                        write!(f, "[")?;
-                        if let Some(rc) = fast {
-                            // sort the elements, because we want the Display output to be
-                            // deterministic, particularly for tests which check equality
-                            // of error messages
-                            for (i, item) in rc.as_ref().iter().sorted_unstable().enumerate() {
-                                write!(f, "{item}")?;
-                                if i < n - 1 {
-                                    write!(f, ", ")?;
-                                }
-                            }
-                        } else {
-                            // don't need to sort the elements in this case because BTreeSet iterates
-                            // in a deterministic order already
-                            for (i, item) in authoritative.as_ref().iter().enumerate() {
-                                write!(f, "{item}")?;
-                                if i < n - 1 {
-                                    write!(f, ", ")?;
-                                }
-                            }
+                write!(f, "[")?;
+                if let Some(rc) = fast {
+                    // sort the elements, because we want the Display output to be
+                    // deterministic, particularly for tests which check equality
+                    // of error messages
+                    for (i, item) in rc.as_ref().iter().sorted_unstable().enumerate() {
+                        write!(f, "{item}")?;
+                        if i < authoritative.len() - 1 {
+                            write!(f, ", ")?;
+                        }
+                    }
+                } else {
+                    // don't need to sort the elements in this case because BTreeSet iterates
+                    // in a deterministic order already
+                    for (i, item) in authoritative.as_ref().iter().enumerate() {
+                        write!(f, "{item}")?;
+                        if i < authoritative.len() - 1 {
+                            write!(f, ", ")?;
                         }
-                        write!(f, "]")?;
-                        Ok(())
                     }
-                    n => write!(f, "<set with {} elements>", n),
                 }
+                write!(f, "]")?;
+                Ok(())
             }
             Self::Record(record) => {
-                write!(f, "<first-class record with {} fields>", record.len())
+                write!(f, "{{")?;
+                for (i, (k, v)) in record.as_ref().iter().enumerate() {
```

Comment:
Nit: This looks nicer in the general case, but it's shown as `` `{}` `` in the case with 0 fields. In that case, I'm not sure the help message is so useful (might take a while to realize it's an empty context).